provider/subs/opensubs/src/main/java/butter/droid/provider/subs/opensubs/OpenSubsProvider.java

```diff
@@ -81,23 +78,16 @@ protected OpenSubsProvider(final OpenSubsService service, final Context context)
     }
 
     private Single<List<Subtitle>> fetchSubtitles(@NonNull final Media media) {
-        return fetchSubtitles(media, QuerySearchRequest.LANGUAGE_ALL);
+        return fetchSubtitles(media, "all");
     }
 
     private Single<List<Subtitle>> fetchSubtitles(@NonNull final Media media, String languageCode) {
-        // TODO cache token
-        return service.login(new String[]{"", "", Locale.getDefault().getLanguage(), USER_AGENT}) // TODO add constants
-                .flatMap((Function<LoginResponse, SingleSource<SearchResponse>>) loginResponse -> {
-                    List<Object> params = new ArrayList<>();
-                    params.add(loginResponse.getTokem());
-                    // TODO add imdb id search
-                    params.add(Collections.singletonList(new QuerySearchRequest(media.getTitle(), languageCode)));
-
-                    return service.search(params);
-                })
-                .map(SearchResponse::getData)
-                .flatMapObservable(Observable::fromIterable)
-                .groupBy(OpenSubItem::getLanguageCode)
+        String imdbid = media.getId().replaceAll("^tt", "");
```

Comment:
We coul'd use it if present. I was also thinking a special search for episodes with season number, episode number and imdb code as params. 
It would need to check the type of the Media instance to see if it is a Movie or an Episode.

Reply:
If the imdb id is not present, then use another search option, with the query as string

Reply:
if you merge in develop you will see that i have added meta data to media. And that one will contain imdb id if it is present. So you should use that.